### DISCUSSION TITLE
Why is the -r flag testing every file in the folder?

### DISCUSSION THREAD

**Discussion by samuelwhiskeyjohnson** [pytest-dev/pytest#12309] [Q&A]
I just want to test `test_skip.py` file.

But `pytest -r test_skip.py` is testing every file in the folder that contains other .py files.

<img width="674" alt="image" src="https://github.com/pytest-dev/pytest/assets/71688370/7e4874f3-1883-4db3-a048-df61da035f3d">

**Comment by RonnyPfannschmidt** [comment]
The r flags expects a parameter 

Due to the lack it seems like argparse considers the filename as argumr to the flag and takes it off the list of files 

Please add Arguments to `-r`
